reporter/sender-kinesis/src/test/java/zipkin2/reporter/kinesis/KinesisSenderTest.java

```diff
@@ -64,105 +57,54 @@ void setup() {
             .build();
   }
 
-  @Test
-  void sendsSpans() throws Exception {
+  @Test void send() throws Exception {
     server.enqueue(new MockResponse());
 
-    send(CLIENT_SPAN, CLIENT_SPAN).execute();
+    sendSpans(CLIENT_SPAN, CLIENT_SPAN);
 
     assertThat(extractSpans(server.takeRequest().getBody()))
         .containsExactly(CLIENT_SPAN, CLIENT_SPAN);
   }
 
-  @Test
-  void sendsSpans_PROTO3() throws Exception {
+  @Test void send_empty() throws Exception {
     server.enqueue(new MockResponse());
 
-    sender.close();
-    sender = sender.toBuilder().encoding(Encoding.PROTO3).build();
-
-    send(CLIENT_SPAN, CLIENT_SPAN).execute();
+    sendSpans();
 
     assertThat(extractSpans(server.takeRequest().getBody()))
-        .containsExactly(CLIENT_SPAN, CLIENT_SPAN);
+        .isEmpty();
   }
 
-  @Test
-  void outOfBandCancel() throws Exception {
+  @Test void send_PROTO3() throws Exception {
     server.enqueue(new MockResponse());
 
-    KinesisSender.KinesisCall call = (KinesisSender.KinesisCall) send(CLIENT_SPAN, CLIENT_SPAN);
-    assertThat(call.isCanceled()).isFalse(); // sanity check
-
-    CountDownLatch latch = new CountDownLatch(1);
-    call.enqueue(new Callback<>() {
-      @Override
-      public void onSuccess(Void aVoid) {
-        call.future.cancel(true);
-        latch.countDown();
-      }
-
-      @Override
-      public void onError(Throwable throwable) {
-        latch.countDown();
-      }
-    });
-
-    latch.await(5, TimeUnit.SECONDS);
-    assertThat(call.isCanceled()).isTrue();
+    sender.close();
+    sender = sender.toBuilder().encoding(Encoding.PROTO3).build();
+
+    sendSpans(CLIENT_SPAN, CLIENT_SPAN);
+
+    assertThat(extractSpans(server.takeRequest().getBody()))
+        .containsExactly(CLIENT_SPAN, CLIENT_SPAN);
   }
 
-  @Test
-  void sendsSpans_json_unicode() throws Exception {
+  @Test void send_json_unicode() throws Exception {
     server.enqueue(new MockResponse());
 
     Span unicode = CLIENT_SPAN.toBuilder().putTag("error", "\uD83D\uDCA9").build();
-    send(unicode).execute();
+    sendSpans(unicode);
 
     assertThat(extractSpans(server.takeRequest().getBody())).containsExactly(unicode);
   }
 
-  @Test
-  void checkPasses() throws Exception {
-    enqueueCborResponse(
-        mapper
-            .createObjectNode()
-            .set("StreamDescription", mapper.createObjectNode().put("StreamStatus", "ACTIVE")));
-
-    CheckResult result = sender.check();
-    assertThat(result.ok()).isTrue();
-  }
-
-  @Test
-  void checkFailsWithStreamNotActive() throws Exception {
```

Comment:
this test isn't viable as the actual client never looks for the StreamStatus, so regardless of it it was deleting or not, you would get some other kind of exception. This is a good example of why testing with empty list is more accurate on client experience.